photutils/psf/photometry.py

```diff
@@ -15,6 +15,381 @@
+        these parameters should be given as ``x_0``, ``y_0`` and
+        ``flux``.  `~photutils.psf.prepare_psf_model` can be used to
+        prepare any 2D model to match this assumption.
+    fitshape : int or length-2 array-like
+        Rectangular shape around the center of a star which will be used
+        to collect the data to do the fitting. Can be an integer to be
+        the same along both axes. E.g., 5 is the same as (5, 5), which
+        means to fit only at the following relative pixel positions:
+        [-2, -1, 0, 1, 2].  Each element of ``fitshape`` must be an odd
+        number.
+    finder : callable or instance of any `~photutils.detection.StarFinderBase` subclasses or None
+        ``finder`` should be able to identify stars, i.e. compute a
+        rough estimate of the centroids, in a given 2D image.
+        ``finder`` receives as input a 2D image and returns an
+        `~astropy.table.Table` object which contains columns with names:
+        ``id``, ``xcentroid``, ``ycentroid``, and ``flux``. In which
+        ``id`` is an integer-valued column starting from ``1``,
+        ``xcentroid`` and ``ycentroid`` are center position estimates of
+        the sources and ``flux`` contains flux estimates of the sources.
+        See, e.g., `~photutils.detection.DAOStarFinder`.  If ``finder``
+        is ``None``, initial guesses for positions of objects must be
+        provided.
+    fitter : `~astropy.modeling.fitting.Fitter` instance
+        Fitter object used to compute the optimized centroid positions
+        and/or flux of the identified sources. See
+        `~astropy.modeling.fitting` for more details on fitters.
+    aperture_radius : float
+        The radius (in units of pixels) used to compute initial
+        estimates for the fluxes of sources. If ``None``, one FWHM will
+        be used if it can be determined from the ```psf_model``.
+
+    Notes
+    -----
+    If there are problems with fitting large groups, change the
+    parameters of the grouping algorithm to reduce the number of sources
+    in each group or input a ``star_groups`` table that only includes
+    the groups that are relevant (e.g. manually remove all entries that
+    coincide with artifacts).
+
+    References
+    ----------
+    [1] Stetson, Astronomical Society of the Pacific, Publications,
+        (ISSN 0004-6280), vol. 99, March 1987, p. 191-222.
+        Available at: http://adsabs.harvard.edu/abs/1987PASP...99..191S
+    """
+
+    def __init__(self, group_maker, bkg_estimator, psf_model, fitshape,
+                 finder=None, fitter=LevMarLSQFitter(), aperture_radius=None):
+        self.group_maker = group_maker
+        self.bkg_estimator = bkg_estimator
+        self.psf_model = psf_model
+        self.finder = finder
+        self.fitter = fitter
+        self.fitshape = fitshape
+        self.aperture_radius = aperture_radius
+        self._residual_image = None
```

Comment:
yep, that's the right idea.  An alternative that's functionally equivalent but might be simpler is to have it be a read-only attribute.  I slightly favor that because to me `.get_xxx` implies something needs to be "done", while `.xxx` just means "need to access a variable".

Also, it *might* be better to have the name be `last_residual_image` or something along those lines.  That makes it clearer that it's only from the last time this class was used, and older ones are just lost.